src/Microsoft.Identity.Web.Certificate/SignedAssertionFromManagedIdentityCredentialLoader.cs

```diff
@@ -21,18 +21,14 @@ public async Task LoadIfNeededAsync(CredentialDescription credentialDescription,
                 {
                     managedIdentityClientAssertion = new ManagedIdentityClientAssertion(credentialDescription.ManagedIdentityClientId, credentialDescription.TokenExchangeUrl);
                 }
-                try
-                {
-                    // Given that managed identity can be not available locally, we need to try to get a
-                    // signed assertion, and if it fails, move to the next credentials
-                    _= await managedIdentityClientAssertion!.GetSignedAssertionAsync(null);
-                    credentialDescription.CachedValue = managedIdentityClientAssertion;
-                }
-                catch (AuthenticationFailedException)
```

Comment:
This the root cause of the bug. When we moved from Azure SDK to MSAL, I forgot to update this exception type. Ultimately, it doesn't matter why the credential fails, so a catch all is better used (up the stack in DefaultCredentialLoader)

Reply:
Makes sense. thanks for finding that out.

Reply:
That explains why this showed up recently then.